back/engines/free/email_campaigns/app/models/email_campaigns/campaigns/comment_deleted_by_admin.rb

```diff
@@ -68,6 +69,20 @@ def generate_commands(recipient:, activity:, time: nil)
       }]
     end
 
+    def activity_context(activity)
+      return nil unless activity.item.is_a?(::Notification)
+
+      activity.item.idea && TimelineService.new.current_phase(activity.item.idea)
```

Comment:
This confused me for some time. Maybe we should rename the parameter in the TimelineService method, to clarify that the project can be deduced from objects other than a project?

Maybe `def current_phase(item, time = Time.now)` is better?

```Ruby
def current_phase(item, time = Time.now)
    date = tenant_timezone.at(time).to_date
    item.phases.find do |phase|
      phase.start_at <= date && (phase.end_at.nil? || phase.end_at >= date)
    end
  end
  ```

Reply:
I didn't realize this and just copy-pasted something like `TimelineService.new.current_phase(idea)` from somewhere else 😅

Agreed. Perhaps, we could have three methods: `current_phase(phases)`, `current_idea_phase(idea)` and `current_project_phase(project)`? The latter two would call the first one.

Reply:
Seems like the place I copied this from actually passed a project. So I'm now making sure we're always passing a project.